### ISSUE TITLE
Support for PostgreSQL SCRAM-SHA-256 or even generic SCRAM-SHA-256

### ISSUE THREAD

**Issue by clem9669** [hashcat/hashcat#3038]
**Is your request based on a publicly known cryptographic schema and where can we find information about?**
PostgreSQL has a SCRAM-SHA-256 password authentication as describe: https://www.postgresql.org/docs/13/auth-password.html
The RFC for this algorithm is https://datatracker.ietf.org/doc/html/rfc7677.

**Describe alternatives you've considered**
The support for this algorithm has already discussed a bit in https://github.com/hashcat/hashcat/issues/2551
Instead of one mode per implementation, it would be great to have a generic SCRAM-SHA-256 mode as explained by  Emvee321 in https://github.com/hashcat/hashcat/issues/2551#issuecomment-703086152

**Provide an example hash/database/file and a known correct password**
hashcat1:123hashcat*
```sh
postgres=# SELECT rolpassword FROM pg_authid where rolname = 'hashcat1';
             rolpassword                                                              
---------------------------------------------------------------------------------------------------------------------------------------
 SCRAM-SHA-256$4096:gyK5c548WpO1JDAel6CEhQ==$n5+OPFYnUaoS6x+vcjIsA3YxliDHQBmHA5+5n0RsWTg=:YKvnq79aglw18SfGBiaHmPrAiyzuMpTnPrCAuoruB3s=
(1 row)
```

**Explain how the cryptographic schema is implemented**
The following Cyberchef recipe demonstrates the SCRAM-SHA256 scheme for PostgreSQL (only the number of iterations (4096) differ from MongoDB (15000).
Use the following two lines of input (password and base64 salt (see example postgreSQL database hash string for salt).
```sh
123hashcat
gyK5c548WpO1JDAel6CEhQ==
```
```sh
Register('([\\s\\S]*)\\n([\\s\\S]*)',true,false,false)
Find_/_Replace({'option':'Regex','string':'([\\s\\S]*)\\n([\\s\\S]*)'},'$R0',true,false,true,false)
Derive_PBKDF2_key({'option':'UTF8','string':'$R0'},256,4096,'SHA256',{'option':'Base64','string':'$R1'})
Register('([\\s\\S]*)',true,false,false)
Find_/_Replace({'option':'Regex','string':'$R2'},'Server Key',true,false,true,false)
HMAC({'option':'Hex','string':'$R2'},'SHA256')
From_Hex('Auto')
To_Base64('A-Za-z0-9+/=')
```
https://gchq.github.io/CyberChef/#recipe=Register('(%5B%5C%5Cs%5C%5CS%5D*)%5C%5Cn(%5B%5C%5Cs%5C%5CS%5D*)',true,false,false)Find_/_Replace(%7B'option':'Regex','string':'(%5B%5C%5Cs%5C%5CS%5D*)%5C%5Cn(%5B%5C%5Cs%5C%5CS%5D*)'%7D,'$R0',true,false,true,false)Derive_PBKDF2_key(%7B'option':'UTF8','string':'$R0'%7D,256,4096,'SHA256',%7B'option':'Base64','string':'$R1'%7D)Register('(%5B%5C%5Cs%5C%5CS%5D*)',true,false,false)Find_/_Replace(%7B'option':'Regex','string':'$R2'%7D,'Server%20Key',true,false,true,false)HMAC(%7B'option':'Hex','string':'$R2'%7D,'SHA256')From_Hex('Auto')To_Base64('A-Za-z0-9%2B/%3D')&input=MTIzaGFzaGNhdApneUs1YzU0OFdwTzFKREFlbDZDRWhRPT0

This issue content is mostly from @Emvee321. @philsmd asked for separe issue (https://github.com/hashcat/hashcat/issues/2551#issuecomment-703217168)

**Comment by Neustradamus** [comment]
@ all: I wish you a Happy New Year 2022!

Any news about it?

**Comment by b0lek** [comment]
I've created PR #3138  for adding this new module.

**Comment by matrix** [comment]
closed by #3138
thanks :)

**Comment by Neustradamus** [comment]
Thanks @b0lek for the PR and @jsteube for merging! :)

Thanks @clem9669, @philsmd, @matrix too.

Linked to:
- https://github.com/scram-sasl/info/issues/1
